File: 2021_May/4.cpp
class Solution {
public:
    bool checkPossibility(vector<int>& nums) {
        int cnt = 1, n = nums.size();
        for (int i = 1; i < n; ++i) {
            if (nums[i] < nums[i - 1]) {
                if (cnt == 0) return false;
                if (i == 1 || nums[i] >= nums[i - 2]) nums[i - 1] = nums[i];
                else nums[i] = nums[i - 1];
                --cnt;
            } 
        }
        return true;
    }
};

File: 2022_Feb/4.py
class Solution:
    def findMaxLength(self, nums: List[int]) -> int:
        dict = {0:-1}
        s = 0
        res = 0
        for i in range(len(nums)):
            s += 1 if nums[i] == 1 else -1
            if (s in dict):
                res = max(res, i - dict[s])
            else:
                dict[s] = i
        #print(dict)
        return res
